### ISSUE TITLE
External Touch Screens Improperly Registering Touches on Steam Deck OS 3.5.5+

### ISSUE THREAD

**Issue by Jsalinger** [ValveSoftware/SteamOS#1259]
#### Your system information

* Steam client version: 1701289036
* SteamOS version: 3.5.5 and 3.5.7
* Opted into Steam client beta?: [Yes/No]  No
* Opted into SteamOS beta?: [Yes/No]  No
* Have you checked for updates in Settings > System?: [Yes/No] Yes

#### Please describe your issue in as much detail as possible:
External touch screen monitor not working properly when connected directly via USB-C to Steam Deck on OS versions 3.5.5 and 3.5.7. 

This issue example is with an XTendtouch XT1610F. The touch screen works perfectly with Steam Deck OS 3.4.11, but on 3.5.5 the touch screen recognizes touches, but they are not registered in the correct location on the screen (x,y position is incorrect). Touches are also laggy and generally unresponsive. 

I was able to replicate and remedy this behavior several times, by switching between 3.4.11 (working perfectly), and back to 3.5.5 (not working) on an LCD Deck.

I also confirmed the same broken external touch screen behavior on 3.5.7 on an OLED deck.

Note: other users are also reporting similar behavior in this reddit thread: https://www.reddit.com/r/SteamDeck/comments/17y7dvd/touch_input_on_external_touchscreen_wrong_way/?utm_source=share&utm_medium=web2x&context=3

#### Steps for reproducing this issue:

1. Plug in a USB-C cable directly from the Steam Deck to the external monitor with either Steam Deck OS 3.5.5 or 3.5.7 on either an LCD Deck or an OLED Deck.
2.  Wait for monitor to display screen (screen looks normal).
3.  Proceed to use any touch input on the external screen, and note that scrolling is unresponsive, and when clicks do register, they are for the wrong x,y position.
4. To remedy the issue, revert the OS on either Steam Deck LCD or OLED to 3.4.11, and all touch inputs work perfectly.

**Comment by RandomSynapse** [comment]
Also encountered on Lenovo M14t and an original Steam Deck running the latest betas:

-     Steam client version: 1702667398
-     SteamOS version: 3.5.11 (Build 20231213.2)
-     Opted into Steam client beta?: [Yes/No] Yes
-     Opted into SteamOS beta?: [Yes/No] Yes
-     Have you checked for updates in Settings > System?: [Yes/No] Yes

**Comment by AlexFolland** [comment]
This is happening to me, except with the internal Steam Deck display's touchscreen.  I have both a Steam Deck and a Steam Deck OLED, and this issue is only happening on the Steam Deck (non-OLED), and only in desktop mode.  The issue is not happening in gaming mode.  In desktop mode, the Steam Deck's display is the right way up, as configured with in the KDE Plasma settings with resolution 800x1280 and orientation sideways (the option on the far right).  The image is correctly aligned with the device, but the touchscreen input position mapping is shifted by 90 degrees.  This causes a tap near the top-middle of the screen to click with the mouse cursor near the left-middle of the screen instead.  It maps top-left touches to bottom-left as well, and so on.

I am opted into beta on both devices, with the latest beta installed as of posting this comment.

**Comment by RaBa85** [comment]
Have the same issue but only in "Game Mode" it is working fine in "Desktop Mode", same behavior on Stable and Beta on Steam Deck LCD

Steam client version: 1710458266 & 1713317598
SteamOS version: 3.5.17
Opted into Steam client beta?: [Yes/No] No
Opted into SteamOS beta?: [Yes/No] Yes & No
Have you checked for updates in Settings > System?: [Yes/No] Yes

**Comment by KamilLenica** [comment]
The same issue with new dock. Desktop mode works fine

**Comment by JKlessens** [comment]
I have the same issue with the steam deck oled and a Uperfect portable touchscreen display. Both in game and desktop mode. Pretty sure it has something to do with the strange orientation of the internal display messing with the touch screen of the external display. 
I have read on reddits that this was not an issue in version 3.4.x. It only became an issue when some kind of orientation issue was fixed in 3.5.x. (unconfirmed)
Also I read in two posts that it wasn't an issue when running windows (unconfirmed)

**Comment by pastafizia** [comment]
I have the same problem with a new Steam Deck OLED and a portable ASUS ZenScreen Ink... I really need this setup to work, since I'm replacing my PC with this...

Steam client version: 17s
SteamOS version: 3.5.11 (Build 20231213.2)
Opted into Steam client beta?: [Yes/No] Yes
Opted into SteamOS beta?: [Yes/No] Yes
Have you checked for updates in Settings > System?: [Yes/No] Yes

**Comment by vladkosarev** [comment]
Same issue here with an external 4k OLED monitor.

**Comment by twisted-space** [comment]
Same issue with touch being rotated 90% here but only in game mode. In desktop mode everything works as expected.

Steam client version: 1721173382
SteamOS version: 3.5.19 
Opted into Steam client beta?: [Yes/No] No
Opted into SteamOS beta?: [Yes/No] No
Have you checked for updates in Settings > System?: [Yes/No] Yes

**Comment by Eisa01** [comment]
For me touch input seems totally off.
The issue happens in both Desktop mode and Gaming mode, using 7-inch touch 1080p external display. 

Steam Deck Variant: Steam Deck OLED.
Steam Client Version: 1726683985
Steam OS Version: 3.5.19
Opted into Steam client beta?: [Yes/No]: No
Opted into SteamOS beta?: [Yes/No]: No
Have you checked for updates in Settings > System?: [Yes/No]: Yes

**Comment by astromediaonly** [comment]
3.6.20 too - LCD SteamDeck, bought portable 4k oled touchscreen to create portable setup and when I click on back button in game mode it opens profile which is button on other side of the display ... In Desktop mode it works fine as long as that touchscreen is the only display (internal one is turned off) else the linux for some reason seems to map touchscreen input from one display to whole display area (both displays comibned) so its off by size/resolution of the internal screen.

NOTE: Steam's onscreen keyboard is also unusable in touch monitor setup because touching it removes focus from controls you need to write in
PS: I would appretiate also MPP Pen support (specifically ability to map its buttons to right click in desktop mode) but I know its kinda niche (mentioning it mostly if some expert know how to make it work please contact me :) )

**Comment by chainlink13** [comment]
This issue is still occuring with the latest steam beta.

**Comment by abisov** [comment]
+1

**Comment by sarcysoft** [comment]
This is still an issue as of OS 3.7.1

**Comment by JHCC007** [comment]
Same issue with a portable monitor

**Comment by crazyrokr** [comment]
The same issue with Eviciv OLED 16" 3K

Steam client version: 1745876290
SteamOS version: 3.6.24
Opted into Steam client beta?: No
Opted into SteamOS beta?: No
Have you checked for updates in Settings > System?: Yes

**Comment by mebalzer** [comment]
Using SteamOS 3.8 on Minisforum EM680 (World's Tiniest Steam Box) and it completely reboots the GameScope Client. Works great in Desktop.  This will need to be fixed before this becomes an OS for standard PCs.

**Comment by PJDiddy** [comment]
Bump! Could someone please fix this, it's been nearly 2 years now and I'm still having to run my Deck on 3.4.11!

**Comment by prestonsmartuk** [comment]
Same issue here! I'm amazed this still hasn't been resolved after all this time

**Comment by astromediaonly** [comment]
Yeah, why even bother to report issues / allow us to do so if there isnt even (after several years) indication Valve even reads those :/ ...

**Comment by paopao-zcy** [comment]
same in 3.7.13, touch can not work properly on external monitor.

**Comment by PJDiddy** [comment]
> This should be fixed with [2c92d6e](https://github.com/ValveSoftware/gamescope/commit/2c92d6e15d66a6442091b86fff8efca12275e84d), and hopefully part of the upcoming SteamOS 3.8 release.

I saw that 3.8 entered the preview branch, do you know this fix was included?

**Comment by kigyo** [comment]
> > This should be fixed with [2c92d6e](https://github.com/ValveSoftware/gamescope/commit/2c92d6e15d66a6442091b86fff8efca12275e84d), and hopefully part of the upcoming SteamOS 3.8 release.
> 
> I saw that 3.8 entered the preview branch, do you know this fix was included? 

Can confirm, it works for me now!

**Comment by JJ6981** [comment]
I'm on 3.8.9 and the external monitor is now receiving touch input correctly, but the built-in monitor is not. All of the built-in  monitors inputs are only outputting to my external touchscreen monitor.
